devices/SPS30/Extensions/SerialPortExtensions.cs

```diff
@@ -0,0 +1,93 @@
+﻿//
+// Copyright (c) 2017 The nanoFramework project contributors
+// See LICENSE file in the project root for full license information.
+//
+
+using System.Diagnostics;
+using System.Threading;
+
+namespace System.IO.Ports
+{
+    /// <summary>
+    /// Extensions on the SerialPort to facilitate unstuffing and reading reply frames.
+    /// </summary>
+    public static class SerialPortExtensions
+    {
+        /// <summary>
+        /// Attempts to read an SHDLC frame, which has a 0x7e start/end bytes with stuffing/escaping in between. The
+        /// returned frame will be unstuffed in the process.
+        /// </summary>
+        public static byte[] ReadUnstuffedFrameContent(this SerialPort serialPort)
+        {
+            var sw = Stopwatch.StartNew();
+            using var frame = new MemoryStream();
+            using var rawdata = new MemoryStream();
+            try
+            {
+
+                // Read until frame end byte
+                while (true)
+                {
+                    // This timeout check is meant to activate for situations where garbage keeps coming in at a steady pace within the ReadTimeout, possibly causing
+                    // an infinite loop.
+                    if (serialPort.ReadTimeout > 0 && sw.ElapsedMilliseconds > serialPort.ReadTimeout)
+                        throw new ApplicationException("Timeout waiting for frame content or end byte (0x7e)");
+
+                    // Turns out sleeping works better than ReadByte() when we're waiting for data (nanoframework bug?)
+                    if (serialPort.BytesToRead == 0)
+                    {
+                        Thread.Sleep(5);
+                        continue;
+                    }
+                    var b = (byte)serialPort.ReadByte();
```

Comment:
for readability, we always add a line after closing brackets when they are not followed by another closing one or an else sentence or equivalent.
So please add a line. There may be few others there and there (I'll try to catch them up)